common/src/main/java/org/hawkular/accounts/common/TokenVerifier.java

```diff
@@ -32,11 +39,41 @@
     @Inject @RealmName
     private String realm;
 
+    @Inject @HostSynonyms
+    private Set<String> hostSynonyms;
+
     @Inject
     AuthServerRequestExecutor executor;
 
     public String verify(String token) throws Exception {
-        String tokenUrl = baseUrl
+        JWSInput jwsInput;
+        try {
+            jwsInput = new JWSInput(token);
+        } catch (Exception e) {
+            throw new VerificationException("Couldn't parse token", e);
+        }
+
+        AccessToken accessToken;
+        try {
+            accessToken = jwsInput.readJsonContent(AccessToken.class);
+        } catch (IOException e) {
+            throw new VerificationException("Couldn't parse token signature", e);
+        }
+
+        URL backendUrl = new URL(accessToken.getIssuer());
+        URL baseUrlToCall = new URL(baseUrl);
+        if (!backendUrl.getHost().equalsIgnoreCase(baseUrlToCall.getHost())) {
+            if (hostSynonyms.contains(backendUrl.getHost())) {
```

Comment:
Does that really include text versions of the hostname? e.g. my box has a local ip of 1.2.3.4 with name bla. Would this
catch http://bla:8080/  or (even http://bla.acme.org:8080) if bla.acme.org would map to 1.2.3.4 ?